velox/functions/lib/Re2Functions.h

```diff
@@ -197,4 +198,48 @@ struct Re2RegexpReplace {
   }
 };
 
+template <typename T, std::string (*prepareRegexpPattern)(const StringView&)>
+struct Re2RegexpSplit {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  static constexpr int32_t reuse_strings_from_arg = 0;
+
+  std::optional<RE2> re_;
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const core::QueryConfig& /*config*/,
+      const arg_type<Varchar>* /*string*/,
+      const arg_type<Varchar>* pattern) {
+    VELOX_USER_CHECK(
+        pattern != nullptr, "Pattern of regexp_split must be constant.");
+
+    auto processedPattern = prepareRegexpPattern(*pattern);
+
+    re_.emplace(processedPattern, RE2::Quiet);
+    if (UNLIKELY(!re_->ok())) {
+      VELOX_USER_FAIL(
+          "Invalid regular expression {}: {}.", processedPattern, re_->error());
+    }
+  }
+
+  FOLLY_ALWAYS_INLINE bool call(
+      out_type<velox::Array<Varchar>>& out,
+      const arg_type<Varchar>& string,
+      const arg_type<Varchar>& /*pattern*/) {
+    re2::StringPiece input(string.data(), string.size());
+    re2::StringPiece match;
+    size_t lastEnd = 0, offset = 0;
+    while (re_->Match(
+        input, offset, input.length(), RE2::Anchor::UNANCHORED, &match, 1)) {
+      size_t length = match.data() - input.data() - lastEnd;
+      out.add_item().copy_from(StringView(input.data() + lastEnd, length));
```

Comment:
Should I be using `copy_from` or `setNoCopy` ? I understood `copy_from` doesn't actually copy here https://github.com/facebookincubator/velox/blob/dfad409dee702f1f91ce5d9e87abc259e5e03e92/velox/examples/SimpleFunctions.cpp#L296-L299